msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AppServiceManagedIdentity.java

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+class AppServiceManagedIdentity extends AbstractManagedIdentitySource{
```

Comment:
Is App Service managed identity implementation that different that it warrants its own code path?

Reply:
Yeah, that's the worse part of MSI - there are 5 different implemenations depending on the compute used - Imds, ServiceFabric, AppService, Arc and CloudShell. 

And the logic to figure out which compute we're on is both brittle and slow, as it's based on reading a bunch of env variables and on making local HTTP calls. 


But my understanding is that with SLC, this will more or less be fixed. @gladjohn to keep me honest.